dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/utils/PayloadValidator.kt

```diff
@@ -0,0 +1,49 @@
+package org.dataland.datalandcommunitymanager.utils
+
+import org.dataland.datalandbackend.openApiClient.model.DataTypeEnum
+import org.dataland.datalandcommunitymanager.model.elementaryEventProcessing.ElementaryEventPayloadMetaInfo
+import org.dataland.datalandmessagequeueutils.exceptions.MessageQueueRejectException
+import org.json.JSONObject
+import java.util.*
+
+/**
+ * Validator to check the contents of payloads in the message queue
+ */
+object PayloadValidator {
```

Comment:
Das ist nicht so ganz gelungen. Das validieren und extrahieren sollte nicht in der gleichen Funktion passieren. Ich würde einen payload handler vorschlagen der eine Funktion validate und eine Funktion extractMetaInfor hat oder sowas in der Art.

Reply:
Lass uns nochmal kurz hierüber sprechen.

Reply:
Wurde jetzt gesplittet.

Es gibt eine Funktion zum validaten, und eine weitere welche versucht die payload in ein ElementaryEventBasicInfo-Objekt zu parsen.